### DISCUSSION TITLE
Raytracing for powder bed absorptivity calculations

### DISCUSSION THREAD

**Discussion by B09di** [idaholab/moose#20098] [Q&A General] [answered]
Hello MOOSE community.  

I'm interested in implementing raytracing in MOOSE to calculate Absorptivity of powder bed in the context of Laser Powder Bed Fusion/Selective Laser Sintering  for some configuration/microstructure I have which uses diffuse interface. If anyone has a clue how to implement it in MOOSE, please share the idea or an input file for a similar problem. Any help is highly appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

How is the powder going to be represented in the model exactly?

Guillaume

**Reply by B09di** [reply]
Hello Guillaume. Thank you for your response.

First of all, we generated powder bed using GeoDict allowing for point contact between the particles. We then ran PhaseField sintering simulations on it to allow for some neck growth between the particles. we have the resultant microstructure for which we want to determine the absorptivity. We want to represent the resultant microstructure/powder bed using unstructured finite element mesh in the model.

**Reply by GiudGiud** [reply]
so you have an explicitly heterogeneous geometry?
and you have volumetric properties for laser absorption, and reflectivity for the surfaces?

how big is the laser compared to the particles?

**Reply by GiudGiud** [reply]
@B09di any news on this?

**Reply by liang-tech** [reply]
I have the same idea as B09di. The powder bed has many particles/spheres that can absorb  the energy of laser and reflect the laser. In phase field, we can use an order parameter c to represent solid/particle (c =1) and gas (c = 0). Therefore, the laser can transport in the gas and be absorbed by the solid surface. For the question "how big is the laser compared to the particles?", the particles and the laser have the same radius. The laser beam should be splited into many parts due to heterogeneity power.

**Reply by GiudGiud** [reply]
so you are envisioning this as a ray tracing problem? Or should you be capturing diffraction?

**Reply by liang-tech** [reply]
I think it is more like a problem about geometrical optics. We need to track the path of laser beam and find the area absorbing the energy from laser. (Reference:  Zhou Xu, Wang Ze-kun, Liu Peng Hu Mou-bin. Discrepancies between Gaussian surface heat source model and ray tracing heat source model for numerical simulation of selective laser melting. Comput Mech 2022. Doi: 10.1007/s00466-022-02235-1.)

**Reply by GiudGiud** [reply]
If you can make it work with ray tracing or finite elements feel free to try. Otherwise I think moose won't be the right tool
